### PR TITLE
AES-GCM: GMULT using 4-bit table

### PR DESCRIPTION
**Pull request by SparkiDev**
When 64-bit data type available and not big endian code is faster.
--enable-aesgcm=4bit

### CODE REVIEW

File: wolfcrypt/src/misc.c [hunk 1]
```diff
@@ -131,6 +131,13 @@ WC_STATIC WC_INLINE word32 ByteReverseWord32(word32 value)
     /* 5 instructions with rotate instruction, 9 without */
     return (rotrFixed(value, 8U) & 0xff00ff00) |
            (rotlFixed(value, 8U) & 0x00ff00ff);
+#elif defined(WOLFSSL_ARM)
```
Comment (kojo1) [inline wolfcrypt/src/misc.c:134]: Would it fit with WOLFSSL_ARMASM?

Comment (SparkiDev) [reply wolfcrypt/src/misc.c]: Thanks Kojo!
I meant to fix this to be turned on for ARM compilation regardless of WOLFSSL_ARMASM.
Fixed it to detect the CPU based on __arm__ and __aarch64__.

Comment (kojo1) [reply wolfcrypt/src/misc.c]: An user reports that other compiler like Diab VxWorks seems to have direct of inline assembler, and gets error at this line. It maight be good to separate switches for architechture and asm.

Comment (SparkiDev) [reply wolfcrypt/src/misc.c]: OK. Added check for __GNUC__ now before using assembly code.

File: mcapi/crypto.h [hunk 2]
```diff
@@ -167,7 +167,7 @@ enum {
 
 /* AES */
 typedef struct CRYPT_AES_CTX {
-    int holder[90];   /* big enough to hold internal, but check on init */
+    int holder[602];   /* big enough to hold internal, but check on init */
```
Comment (toddouska) [inline mcapi/crypto.h:170]: This is almost a factor of 7 size increase.  Why is this needed for microchip APIs?

Comment (SparkiDev) [reply mcapi/crypto.h]: For some reason I had MS CAPI in my mind.
Change the AES-GCM default for MCAPI to be the non-table version and restored the size.

**Review by toddouska — APPROVED** [review]
